specification/cognitiveservices/data-plane/FormRecognizer/preview/v2.0/FormRecognizer.json

```diff
@@ -0,0 +1,1524 @@
+          }
+        },
+        "valueObject": {
+          "description": "Dictionary of named field values.",
+          "type": "object",
+          "additionalProperties": {
+            "$ref": "#/definitions/FieldValue"
+          }
+        },
+        "text": {
+          "description": "Extracted text content of the recognized field.",
+          "type": "string"
+        },
+        "boundingBox": {
+          "description": "Bounding box of the field text, if appropriate.",
+          "$ref": "#/definitions/BoundingBox"
+        },
+        "confidence": {
+          "description": "Qualitative confidence measure.",
+          "$ref": "#/definitions/Confidence"
+        },
+        "elements": {
+          "description": "List element references.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ElementReference"
+          }
+        }
+      }
+    },
+    "FieldValueType": {
+      "type": "string",
+      "description": "Type of field value.",
+      "enum": [
+        "string",
+        "date",
+        "time",
+        "phoneNumber",
+        "number",
+        "integer",
+        "array",
+        "object"
+      ],
+      "x-ms-enum": {
+        "name": "FieldValueType",
+        "modelAsString": false
+      },
+      "x-nullable": false
+    },    
+    "Confidence": {
+      "description": "Confidence value.",
+      "type": "number",
+      "minimum": 0,
+      "maximum": 1
+    },
+    "CopyRequest": {
+      "description": "Contract to initiate a copy request.",
+      "type": "object",
+      "properties": {
+        "targetInformation": {
+          "description": "Information about target subscription.",
+          "type": "object",
+          "properties":{
+            "endpoint": {
+              "description": "Get or set endpoint path.",
+              "maxLength": 2048,
+              "minLength": 0,
+              "type": "string"
+            },
+            "resourceId": {
+              "description": "Get or set resource identifier.",
+              "maxLength": 2048,
+              "minLength": 0,
+              "type": "string"
+            }
+          }
+        },
+        "copyConfig": {
+          "description": "Copy operation options.",
+          "type": "object",
+          "required":[
+            "deleteSource"
+          ],
+          "properties":{
+            "deleteSource": {
+              "description": "Indicate deletion of source model artifacts.",
+              "type":"boolean",
+              "default":false
+            },
+            "supportCrossTenantCopy": {
+              "description": "Indicate copy target tenancy requirement.",
+              "type": "boolean",
+              "default":false
+            },
+            "targetModelMetadata": {
+              "description": "Metadata for copied model to Target",
+              "type": "object",
+              "properties":{
+                "name": {
```

Comment:
We have no way to retrieve the name. GET /custom/models/{modelId} doesn't return a field called "name". 

When we train the model, we also don't have an option to assigne a name. 

Overall I think it's a good idea to assign a name to a model. Maybe we should add in POST /custom/models as part of trainRequest.

Reply:
You are right that customers have currently have no way of associating any information with the models they produce. One thought was that we will let the customers provide a metadata json fragment along with train that we will save in the backend and simply return with the GET calls. This will allow for them to associate "name", "description" or any relevant attribute for their domain/use case while at the same time not adding much cost for us. 

The copy API is another one where such model metadata injection is possible. In this case, the caller will have the ability to rewrite new metadata for the model that is being "created". This may nor may not be the same as the source model's metadata. However, as you noted allowing this pattern in the Copy API alone will not make much sense.

Reply:
@ramparab - If we want to associate metadata with a model, can we update the targetModelMetadata to be something like this. We can also carry forward this pattern to the train API.

modelMetadata": {
              "description": "Metadata for copied model to target",
              "type": "object",
              "required": false
}